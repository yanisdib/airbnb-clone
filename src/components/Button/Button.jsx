import styled from 'styled-components';


export default styled.button`
    --btn-default-color: #FFFFFF;
    --btn-alt-color: #222222;
    --btn-font-size: 16px;
    --btn-font-weight: 500;
    --btn-line-height: 20px;
    --btn-justify-content: flex-start;
    --btn-padding-y: 14px;
    --btn-padding-x: 24px;

    display: flex;
    justify-content: var(--btn-justify-content);
    width: fit-content;
    height: fit-content;
    background: var(${({ variant }) => variant ? '--btn-alt-color' : '--btn-default-color'});
    color: var(${({ variant }) => variant ? '--btn-default-color' : '--btn-alt-color'});
    font-size: var(--btn-font-size);
    font-weight: var(--btn-font-weight);
    line-height: var(--btn-line-height);
    padding: var(--btn-padding-y) var(--btn-padding-x);
    border-radius: 12px;
`;