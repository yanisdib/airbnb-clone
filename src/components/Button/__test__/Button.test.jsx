import { mount } from 'enzyme';
import toHaveStyleRule from 'jest-styled-components';

import Button from '../Button';


describe('Button component', () => {
    let wrapper;

    beforeAll(() => wrapper = mount(<Button>Hello</Button>))

    it('renders correctly without crashing', () => expect(wrapper).toMatchSnapshot())

    it('renders with text content', () => expect(wrapper.text()).toBe('Hello'))

    it('renders with black background if variant props is true', () => {
        const wrapperVariant = mount(<Button variant />);

        expect(wrapperVariant).toHaveStyleRule('background', 'var(--btn-alt-color)')
    })
})