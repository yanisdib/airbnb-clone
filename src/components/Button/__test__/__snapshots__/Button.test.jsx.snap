// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly without crashing 1`] = `
<styled.button>
  <button
    className="jVRXBs"
  >
    Hello
  </button>
</styled.button>
`;
